domain/consensus/model/externalapi/hash.go

```diff
@@ -65,16 +65,13 @@ func (hash DomainHash) String() string {
 }
 
 // ByteArray returns the bytes in this hash represented as a byte array.
-// The hash bytes are cloned, therefore it is safe to modify the resulting array.
 func (hash *DomainHash) ByteArray() *[DomainHashSize]byte {
-	arrayClone := hash.hashArray
-	return &arrayClone
+	return &hash.hashArray
```

Comment:
Which kind of leak does it fix? Is it a temporary leak that is cleaned by the gc or is it permanent?